### ISSUE TITLE
Add docs and upgrade guide for v8

### ISSUE THREAD

**Issue by NoahZinsmeister** [NoahZinsmeister/web3-react#353]

**Comment by myhendry** [comment]
someone wrote a documentation that I think is helpful 

https://hackmd.io/Ykpp1MWLTjixIZG2ZJEShA#Web3React-API-Reference

@wsschwab also posted in the link below

https://github.com/NoahZinsmeister/web3-react/issues/226

**Comment by myhendry** [comment]
I did a simple boilerplate code using only metamask as wallet, default Greeter.sol contract and ethers hardhat. It is deployed successfully on the rinkeby testnet [etherscan transaction](https://rinkeby.etherscan.io/tx/0xb2d5f60b454f3715a9d79103e4525491ac6689da69ba90ccd18c2a5ff2e45c60). It could be useful to those who are trying to use the library

https://github.com/myhendry/boilerplate

**Comment by jacob-abe** [comment]
Hey Noah, being a kind of a newbie, Id like to contribute to the docs. Do you have any sort of reference I can use to get started or do I have to start from scratch by reading the src code?

**Comment by agcty** [comment]
@NoahZinsmeister Can definitely help out with this one, wdyt about Nextra for creating docs?

**Comment by NoahZinsmeister** [comment]
i've been keen to try out nextra @agcty - if you have a vision i'm interested! are you imagining another sibling package in the monorepo that builds into a static site, or something a bit more standalone?

there's a skeleton amount of TSDoc which i was at some point considering using as the basis for docs, but i'm not sure it's the best foundation - it seems hard to expose TSDoc in anything other than IDE autocompletes, so at the moment i'm leaning towards keeping it around only for that use case, and not as a general purpose solution.

**Comment by agcty** [comment]
@NoahZinsmeister Yeah was thinking of another package in the monorepo, think that's the easiest way and nextra would allow us to showcase the functionality directly in the docs which is a neat side effect.

**Comment by NoahZinsmeister** [comment]
sounds good @agcty ! agree that it'll be cool to consume packages in the docs site the same way the example does. just 2 quick notes:

1. given that the docs will be co-located with everything else, we should probably explicitly call out the few areas where the monorepo structure leads to differences in the install/build process relative to a typical consumer app. for example, `react` is installed at the parent level to avoid dependency conflicts, and thus is not in the package.json of the example, but in a consumer application it would obviously need to be. and technically all the `@web3-react/*` dependencies are installed locally via lerna rather than through npm, but i think that's less of a big deal because in the package.json they appear correct, and lerna just does the linking behind the scenes

2. at some point let's figure out hosting/CI/etc. - right now vercel builds the example next.js site against main and PRs, ideally we can do this alongside the docs - or perhaps we can integrate the example into the docs site

**Comment by JorisZierold** [comment]
@NoahZinsmeister Good to see your efforts on this front as well! I would propose to ship a first update of the docs asap, because the current state is far away from being friendly for newcomers. Therefore, focus less on the technicalities and more on a more user-friendly introduction of this valuable project. If there is a specific way to support, I'm here to help.

**Comment by FranRom** [comment]
An upgrading guide should be prioritised IMO, I wasn't able to find any explicit info about it. Let me know if I'm wrong and there is a list of things to check at least.

**Comment by niraj-khatiwada** [comment]
Anyone knows how to connect web3.js in version 8? I have been trying to do that for hours and I don't find any documentation on internet. Help!!

**Comment by Darshan4114** [comment]
Hi @NoahZinsmeister 
I am a web3 newbie looking for good first issues. Can I help with this one?

**Comment by ashutosh887** [comment]
I would like to work on this @NoahZinsmeister
